cmake/thirdparty/SetupCUDA.cmake

```diff
@@ -7,6 +7,12 @@
 # Sanity Checks
 ################################
 
+# Don't create the cuda and cuda_runtime targets if either already exist.
+if (TARGET cuda OR TARGET cuda_runtime)
+    message(FATAL_ERROR "BLT's CUDA targets have already been created.")
```

Comment:
I'm not sure we want a fatal error if any of the targets already exist. Won't this mess up any project built with BLT that imports a dependency built with BLT?

Reply:
Yeah sorry--`SetupCUDA.cmake`, `SetupMPI.cmake`, `SetupThirdParty.cmake` and `SetupHIP.cmake` are all incomplete while I add tests.  I will ping you and mark the PR as ready for review when I make these changes.